d2/src/main/java/com/linkedin/d2/xds/XdsToD2PropertiesAdaptor.java

```diff
@@ -344,7 +475,7 @@ public void onChanged(XdsClient.D2NodeMapUpdate update)
     @Override
     public void onError(Status error)
     {
-      if (_init.compareAndSet(false, true))
+      if (_isInit.get())
```

Comment:
This means that `onError` can now execute multiple times instead of once. Is that intentional? What was the reasoning behind flipping this init logic?

Reply:
right, what we really want to flag is the initial success of fetching the data, until then, onError shouldn't be flipping the flag.